lib/tools/system-calls.js

```diff
@@ -855,26 +861,41 @@ systemCallMethods.reboot = async function reboot (retries = DEFAULT_ADB_REBOOT_R
   });
 };
 
+/**
+ * Restart adb but only if no devices are connected
+ */
+systemCallMethods.restartAdbIfDevicesOffline = async function restartAdbIfDevicesOffline () {
```

Comment:
I don't like this method in in general. Can we instead analyze the output of root/unroot command and perform the necessary actions if it contains some particular keywords/return exit codes?